1 commit

DmitryKirichenko opened the pull request.

DmitryKirichenko committed:
1 commit

DmitryKirichenko committed:
2 commit

DmitryKirichenko committed:
3 commit

SergeyZhitenev commented:
Старайся называть коммиты информативно, так ты упростишь проверку кода другими людьми, добавишь им понимания, что конкретно делает коммит и какие изменения он вносит. Для примера можно посмотреть эти ресурсы https://habr.com/ru/companies/yandex/articles/431432/
https://www.conventionalcommits.org/ru/v1.0.0/
![image](https://github.com/user-attachments/assets/8e5816b2-9f2f-4733-8eac-413ec790fb93)

SergeyZhitenev commented on src/main/java/school/faang/user_service/mapper/UserMapper.java:
```diff
@@ -0,0 +1,13 @@
+import school.faang.user_service.dto.UserDto;
+import school.faang.user_service.entity.User;
+
+@Mapper(componentModel = "Spring")
```
Добавь в строку параметр unmappedTargetPolicy = ReportingPolicy.IGNOR если при маппинге между двумя объектами появятся какие-то неизвестные поля, то маппер не будет кидать ошибку, а будет их игнорировать. В твоем случае такие поля будут точно ( сравни количество полей в UserDto и User )

SergeyZhitenev commented on src/main/java/school/faang/user_service/service/UserService.java:
```diff
@@ -0,0 +1,16 @@
+    private final UserRepository userRepository;
+
+    public User getUserById(long id) {
+        return userRepository.findById(id).orElse(null);
```
Если здесь ты вернешь null в случае отсутствия user, то в EventPerticipationService в 43 строке у тебя в каком-то случае может этот самый null и оказаться, соответственно дальше все посыпется. Лучше возвращать ошибку

ipasynkov-epam replied:
Добавлю, что когда была тема про Optional, то такие кейсы как раз рассматривались. Если мы понимаем, что метод может вернуть нулл (в БД же можем не найти объект) - то возвращаемый тип пусть будет Optional\<User\> - и на этом строить логику)

SergeyZhitenev commented on src/main/java/school/faang/user_service/service/EventParticipationService.java:
```diff
@@ -0,0 +1,55 @@
+    private final EventValidator eventValidator;
+
+    public void registerParticipant(UserDto userDto, EventDto eventDto) {
+        participantProcess(userDto, eventDto, true);
```
По заданию в методе ожидаются (long eventId, long userId). Как будто перемудрил с выносом всей логики в отдельный метод.  Если пользователь не зарегистрирован - регистрируем, если пытается повторно зарегистрироваться - бросаем исключение. Мне кажется не стоит пытаться объединять логику методов registerParticipant и unregisterParticipant в один метод, как раз избавишься от передачи не совсем понятных "true" и "false", + в класс валидации у тебя вынесено по сути по 2 строки из этих методов, может написать их прямо здесь?
![image](https://github.com/user-attachments/assets/67e3ebf4-dd0b-45ab-b57f-6877b4eacc9a)

The-Naku commented on src/main/java/school/faang/user_service/service/EventParticipationService.java:
```diff
@@ -0,0 +1,55 @@
+        List<User> participants = eventParticipationRepository.findAllParticipantsByEventId(eventDto.id());
+
+        return participants.stream()
+                .map(userMapper::toUserDto)
```
в маппере можно создать метод, который превращает List<User> в List<UserDto> и наоборот. Тогда тебе не нужно будет использовать стрим и превращать каждый обьект по отдельности, сможешь сразу смапить лист

The-Naku commented on src/main/java/school/faang/user_service/validator/EventValidator.java:
```diff
@@ -0,0 +1,18 @@
+package school.faang.user_service.validator;
+
+import org.springframework.stereotype.Component;
```
удали пакет и класс validator.  нет смысла создавать целый класс для одного метода  в несколько строк. таким образом мы бессмысленно расширяем кодовую базу  и ухудшаем читаемость

DmitryKirichenko committed:
Поменял логику в EventParticipationService
Добавил unmappedTargetPolicy в UserMapper
Сделал валидацию в UserService
Удалил EventValidator

DmitryKirichenko committed:
Доработал UserMapper
Изменил getParticipants в EventParticipationService

DmitryKirichenko committed:
style fixes

ipasynkov-epam reviewed (commented):
Хорошо. Теперь остались тесты)

DmitryKirichenko committed:
Написал тесты на сервисы
В EventParticipationService добавил вспомогательный метод isParticipantNotRegistered

DmitryKirichenko committed:
Написал тесты на сервисы
В EventParticipationService добавил вспомогательный метод isParticipantNotRegistered

DmitryKirichenko committed:
Добавил mapping в EventParticipationController

DmitryKirichenko committed:
Merge remote-tracking branch 'refs/remotes/origin/Wyvern-master-stream7' into feature-BJS2-40541

# Conflicts:
#	src/main/java/school/faang/user_service/dto/EventDto.java
#	src/main/java/school/faang/user_service/mapper/UserMapper.java
#	src/main/java/school/faang/user_service/service/UserService.java
#	src/test/java/school/faang/user_service/service/UserServiceTest.java

DmitryKirichenko committed:
Убрал конфликты
Доработал контроллер
-Поправил маршруты
-Поправил возвращаемые HTTP-ответы
